grapple/utils.py

```diff
@@ -4,14 +4,26 @@
 from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
 from django.db import connection
 from graphql import GraphQLError
+from wagtail import VERSION as WAGTAIL_VERSION
 from wagtail.models import Site
 from wagtail.search.index import class_is_indexed
-from wagtail.search.models import Query
 
 from .settings import grapple_settings
 from .types.structures import BasePaginatedType, PaginationType
 
 
+if grapple_settings.ADD_SEARCH_HIT:
+    if WAGTAIL_VERSION >= (6, 0):
+        try:
+            from wagtail.contrib.search_promotions.models import Query
+        except ImportError as e:
+            raise ImportError(
+                "wagtail.contrib.search_promotions app is required for Wagtail 6.0+"
```

Comment:
**Open to suggestions here** 

The Query model is only used if `grapple_settings.ADD_SEARCH_HIT` is set to True and an explict error here seemed appropriate to help users when upgrading.

Reply:
Maybe `ImproperlyConfigured` from Django's exceptions would be a better fit than `ImportError`. I would also suggest making the message something like "wagtail.contrib.search_promotions must be installed if grapple_settings.ADD_SEARCH_HIT=True and using Wagtail >= 6.0"

Reply:
Completely agree! thanks both.